src/goto-symex/symex_clean_expr.cpp

```diff
@@ -76,6 +76,11 @@ process_array_expr(exprt &expr, bool do_simplify, const namespacet &ns)
 
     expr = ode.root_object();
 
+    // if we arrive at a "failed_symbol" there is nothing else to be done - it
+    // has void-type and the caller needs to handle this case gracefully
+    if(!expr.type().get(ID_C_is_failed_symbol).empty())
```

Comment:
Suggest making this a utility method exposed by `add_failed_symbols.h`, otherwise lgtm

Reply:
Done.